### PR TITLE
[Stats] Stats use a mod specific folder

### PR DESCRIPTION
**Pull request by Joelrau**
Stats write&read from "<players_folder>/<mod_folder>/<stats_filename>" aka "players/<fs_game>/iw4x.stat"

### CODE REVIEW

**Comment by Joelrau** [comment]
Fixed a bug

File: src/Components/Modules/ServerList.hpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 // This enables version filtering
-#define VERSION_FILTER
+//#define VERSION_FILTER
```
Comment (momo5502) [inline src/Components/Modules/ServerList.hpp:4]: no.

File: src/Components/Modules/Stats.cpp [hunk 2]
```diff
@@ -84,6 +84,13 @@ namespace Components
 		if (fs_game && fs_game->current.string && strlen(fs_game->current.string) && (!strncmp(fs_game->current.string, "mods/", 5) || !strncmp(fs_game->current.string, "mods\\", 5)))
 		{
 			_file = fs_game->current.string + "/"s + file;
+			if (!FileSystem::FileReader(_file).exists())
+			{
+				Scheduler::Once([]() 
+				{
+					Utils::Hook::Call<void(int)>(0x416E10)(0); // Magically fixes everything.
```
Comment (momo5502) [inline src/Components/Modules/Stats.cpp:91]: give that thing a name

File: src/Game/Structs.hpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #pragma once
 
-#define PROTOCOL 0x96
+#define PROTOCOL 0x95
```
Comment (momo5502) [inline src/Game/Structs.hpp:3]: why are you touching that?

File: src/Components/Modules/D3D9Ex.cpp [hunk 4]
```diff
@@ -748,7 +748,7 @@ namespace Components
 	{
 		if (Dedicated::IsEnabled()) return;
 
-		Dvar::Register<bool>("r_useD3D9Ex", false, Game::dvar_flag::DVAR_FLAG_SAVED, "Use extended d3d9 interface!");
+		Dvar::Register<bool>("r_useD3D9Ex", true, Game::dvar_flag::DVAR_FLAG_SAVED, "Use extended d3d9 interface!");
```
Comment (momo5502) [inline src/Components/Modules/D3D9Ex.cpp:751]: nice, i like that.

File: src/Components/Modules/Node.cpp [hunk 5]
```diff
@@ -167,12 +167,18 @@ namespace Components
 
 	void Node::RunFrame()
 	{
-		if (Dedicated::IsEnabled() && Dvar::Var("sv_lanOnly").get<bool>()) return;
+		if (Dedicated::IsEnabled())
+		{
+			bool isServerLoading = Utils::Hook::Call<bool()>(0x40BC50)();
+			if(Dvar::Var("sv_lanOnly").get<bool>() || isServerLoading)
```
Comment (momo5502) [inline src/Components/Modules/Node.cpp:173]: swap these statements. no need to evaluate the dvar if isServerLoading is already true

File: src/Components/Modules/Node.cpp [hunk 6]
```diff
@@ -167,12 +167,18 @@ namespace Components
...
+				return; // don't run if sv_lanOnly or while server is still loading
+		}
 
-		if (!Dedicated::IsEnabled() && Game::CL_IsCgameInitialized())
+		bool CL_IsCgameTrulyInitialized = Utils::Hook::Get<bool>(0xB2BB8B);
+		if (!Dedicated::IsEnabled() && CL_IsCgameTrulyInitialized)
```
Comment (momo5502) [inline src/Components/Modules/Node.cpp:178]: again, swap these or evaluate the address in the if statement

**Review by momo5502 — CHANGES_REQUESTED** [review]

**Comment by Joelrau** [comment]
Oh man I didn't mean to publish these, only the stat stuff 🤓

**Comment by Joelrau** [comment]
Panicked and deleted the repository, but I can make another one if you like those changes.

**Pull request closed without merging**